route/passwordless.js

```diff
@@ -0,0 +1,49 @@
+const _ = require('lodash')
+const joi = require('joi')
+const jwt = require('jsonwebtoken')
+const router = new (require('koa-router'))()
+const uuidv4 = require('uuid/v4')
+
+const error = require('error')
+const passwordlessRepo = require('repo/passwordless')
+const responder = require('middleware/responder')
+const userRepo = require('repo/user')
+const validate = require('middleware/validate')
+const mailer = require('utils/mailer')
+
+function hasExpired (dateValue) {
+  const date = new Date(dateValue)
+  date.setUTCHours(date.getUTCHours() + _.toInteger(process.env.PASSWORDLESS_DURATION))
+  return date < Date.now()
```

Comment:
IMO this is cleaner (and faster):
```js
const exp = dateValue + process.env.PASSWORDLESS_DURATION * DAY // const DAY = 60 * 60 * 1000
return exp < Date.now()
```

Reply:
there's an inherent problem here where the date value from the database is converted to JS Date when received from the database and then it's toStringed in the mapper (need to double check this) and then we create another JS Date object from that.

Reply:
if `dateValue` is (or can be) a string, then just wrap it with `new Date(...)`, but no need for `get/setUTCHours` mutations.

Reply:
I would suggest to define all duration variables like ` PASSWORDLESS_DURATION=1h`, then your function could be:

```js
function hasExpired (dateValue) {
  const exp = new Date(dateValue) + ms(process.env.PASSWORDLESS_DURATION)
  return exp < Date.now()
}
```

where you would `npm i ms --save`.

In this way it is clear what is the unit of time, and it can be easily changed (`PASSWORDLESS_DURATION=20min`).